compiler/rustc_mir_transform/src/gvn.rs

```diff
@@ -1426,6 +1459,26 @@ impl<'body, 'tcx> VnState<'body, 'tcx> {
             }
         }
 
+        // Transmute-then-PtrToPtr can just transmute the original, so long as the
+        // PtrToPtr won't change metadata (and thus the size of the pointer)
+        if let PtrToPtr = kind
+            && let Value::Cast {
+                kind: Transmute,
+                value: inner_value,
+                from: inner_from,
+                to: _inner_to,
+            } = *self.get(value)
+            && self.pointers_have_same_metadata(from, to)
```

Comment:
Can this block be merged with the previous one? Both `kind` fields can just match with `PtrToPtr | Transmute`?